### ISSUE TITLE
FP 1.10.1 04: script doesn't work any longer

### ISSUE THREAD

**Issue by Illegit** [freeplane/freeplane#296]
**Describe the bug**
this script with the previous release was working, but now with the new FP 1.10.1 04 actually doesn't work.
Is it possible to correct it easily?
Thanks
illegit

**Comment by EdoFro** [comment]
which script?

**Comment by Illegit** [comment]
[LinkPathOpenDirectory.zip](https://github.com/freeplane/freeplane/files/8429994/LinkPathOpenDirectory.zip)
OK, this one, thanks.
illegit

**Comment by macmarrum** [comment]
I'd like to avoid diving into the logic of LinkPathOpenDirectory.groovy
Instead, this is how I would propose to approach the problem
```groovy
// based on OpenCurrentMapDirAction.java
import org.freeplane.core.util.Hyperlink
import org.freeplane.features.link.LinkController
import org.freeplane.plugin.script.proxy.ScriptUtils

def node = ScriptUtils.node()
def c = ScriptUtils.c()

if (node.link.file) {
    def uri = node.link.file.parentFile.toURI()
    LinkController.controller.loadHyperlink(new Hyperlink(uri))
} else {
    c.statusInfo = 'the node is missing a link of type "file"'
}

if (node.externalObject.uri) {
    def file = new File(node.externalObject.uri)
    def uri = file.parentFile.toURI()
    LinkController.controller.loadHyperlink(new Hyperlink(uri))
} else {
    c.statusInfo = 'the node is missing an externalObject'
}
```

**Comment by EdoFro** [comment]
I'm using FP 1.9.14 and it didn't work for me neither.

I have some example scripts I was testing for other purposes, but maybe they can help you too.

In my case I wanted that it opens the explorer with the linked file selected

hope it helps,
edo

----

### 1. using java.awt.Desktop

I think this one only works in higher java versions:

```groovy

import java.io.File;
import java.io.IOException;
import java.awt.Desktop;

    Desktop desktop = null;
    // on Windows, retrieve the path of the "Program Files" folder
    
    File file = node.link.file?:(new File(System.getenv("programfiles")))
    
    try {
          if (Desktop.isDesktopSupported()) {
             desktop = Desktop.getDesktop();
             desktop.browseFileDirectory(file);
          }
          else {
             System.out.println("desktop is not supported");
          }
      }
    catch (IOException e){  }

```

----

### 2. using Runtime.exec

```groovy

import java.io.File;
import java.io.IOException;

File f = node.link.file?:(new File(System.getenv("programfiles")))

Process proc

try {
    def comando = "explorer.exe  /select," + f.getAbsolutePath()
    proc = Runtime.getRuntime().exec(comando);
} catch (Exception ex) {
    System.out.println("Error - " + ex);
}

proc.waitFor()

```

----

### 3. using ProcessBuilder().start()

```groovy

import java.io.File;
import java.io.IOException;

File f = node.link.file?:(new File(System.getenv("programfiles")))

Process proc

try {
    def comando = "explorer.exe  /select," + f.getAbsolutePath()
    pBuilder = new ProcessBuilder(comando.split(" "));
    proc = pBuilder.start();
} catch (Exception ex) {
    System.out.println("Error - " + ex);
}

proc.waitFor()

```

**Comment by Illegit** [comment]
Probably it wasn't working because I used java 8 and now I use java 17... 
Thank you Macamarrum and EdoFro I saw all the proposed solutions.
I tried all and I found that this one: 1. using java.awt.Desktop is generating an error, but all of the others are working, 
I'll study them a little bit, currently, I saw that the ones of Edo are opening many links (that I prefer) I think to utilize the 2.runtime. Thanks again.
illegit
